src/Tribe/Process/Queue.php

```diff
@@ -0,0 +1,316 @@
+			'total'      => count( $this->data ),
+			'fragments'  => $fragments_count,
+		) );
+
+		return $this;
+	}
+
+	/**
+	 * Saves the queue data to the database taking max_packet_size into account.
+	 *
+	 * In some instances the serialized size of the data might be bigger than the
+	 * database `max_packet_size`; trying to write all the data in one query would
+	 * make the db "go away...".
+	 * Here we try to read the database `max_packet_size` setting and use that information
+	 * to avoid overloading the query.
+	 *
+	 * @param       string $key
+	 * @param array        $data
+	 *
+	 * @return int The number of fragments the data was split and stored into.
+	 */
+	protected function save_split_data( $key, array $data ) {
+		if ( empty( $data ) ) {
+			return 0;
+		}
+
+		$max_frag_size = $this->get_max_frag_size();
+		// we add a 15% to the size to take the serialization and query overhead into account when fragmenting
+		$serialized_size = ( strlen( utf8_decode( maybe_serialize( $data ) ) ) ) * 1.15;
+		$frags_count     = (int) ceil( $serialized_size / $max_frag_size );
+		$per_frag        = max( (int) floor( count( $data ) / $frags_count ), 1 );
+
+		$split_data = array_chunk( $data, $per_frag );
+
+		if ( empty( $split_data ) ) {
+			return 0;
+		}
+
+		foreach ( $split_data as $i => $iValue ) {
+			$postfix = $i === 0 ? '' : "_{$i}";
+			update_site_option( $key . $postfix, $split_data[ $i ] );
+		}
+
+		return count( $split_data );
+	}
+
+	/**
+	 * Returns the max frag size in bytes.
+	 *
+	 * The bottleneck here is the database `max_packet_size` so we try to read
+	 * it from the database.
+	 *
+	 * @return int The max size, in bytes, of a data fragment.
+	 */
+	protected function get_max_frag_size() {
+		if ( ! empty( $this->max_frag_size ) ) {
+			return $this->max_frag_size;
+		}
+
+		return tribe( 'db' )->get_max_allowed_packet_size();
+	}
+
+	/**
+	 * Sets the maximum size, in bytes, of the queue fragments.
+	 *
+	 * This will prevent the class from trying to read the value from the database.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $max_frag_size
+	 */
+	public function set_max_frag_size( $max_frag_size ) {
+		$this->max_frag_size = $max_frag_size;
+	}
+
+	/**
+	 * Returns the queue unique identifier.
+	 *
+	 * Mind that an id will only be available after saving a queue.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 * @throws RuntimeException if trying to get the queue id before saving it.
+	 */
+	public function get_id() {
+		if ( null === $this->id ) {
+			// not localized as this is a developer-land error
+			throw new RuntimeException( 'Can only get the id of queue after saving it.' );
+		}
+
+		return $this->id;
+	}
+
+	/**
+	 * Overrides the base `dispatch` method to allow for constants and/or environment vars to run
+	 * async requests in sync mode.
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public function dispatch() {
+		if (
+			( defined( 'TRIBE_NO_ASYNC' ) && true === TRIBE_NO_ASYNC )
```

Comment:
Should we use `tribe_is_truthy`? for both?

Reply:
In this specific case, I'd leave the check as is. Both are `developer` options that will be set to boolean or numeric values and I would skip the added overhead of a more complex function.